### ISSUE TITLE
csharp-ls does not work with eglot

### ISSUE THREAD

**Issue by bradprob** [razzmatazz/csharp-language-server#209]
dotnet: 8.0.404
csharp-ls: 0.15.0.0
emacs: GNU Emacs 29.4 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.43, cairo version 1.18.2) of 2024-12-08, modified by Debian

A project is create with `doetnet new console -n test`. Navigated to the file `Program.cs` and started `eglot`. `Eglot` started without error. When typing ```Console.```, no answer is received from the server by ```eglot```. Below the eglot events.
```
[internal] Sun Dec 22 22:21:00 2024:
(:message "Running language server: /home/USER/.dotnet/tools/csharp-ls")
[client-request] (id:1) Sun Dec 22 22:21:00 2024:
(:jsonrpc "2.0" :id 1 :method "initialize" :params
          (:processId 32096 :rootPath "/home/USER/projects/csharp/test/" :rootUri "file:///home/USER/projects/csharp/test" :initializationOptions #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
                                                                                                                                                                  ())
                      :capabilities
                      (:workspace
                       (:applyEdit t :executeCommand
                                   (:dynamicRegistration :json-false)
                                   :workspaceEdit
                                   (:documentChanges t)
                                   :didChangeWatchedFiles
                                   (:dynamicRegistration t)
                                   :symbol
                                   (:dynamicRegistration :json-false)
                                   :configuration t :workspaceFolders t)
                       :textDocument
                       (:synchronization
                        (:dynamicRegistration :json-false :willSave t :willSaveWaitUntil t :didSave t)
                        :completion
                        (:dynamicRegistration :json-false :completionItem
                                              (:snippetSupport :json-false :deprecatedSupport t :resolveSupport
                                                               (:properties
                                                                ["documentation" "details" "additionalTextEdits"])
                                                               :tagSupport
                                                               (:valueSet
                                                                [1]))
                                              :contextSupport t)
                        :hover
                        (:dynamicRegistration :json-false :contentFormat
                                              ["plaintext"])
                        :signatureHelp
                        (:dynamicRegistration :json-false :signatureInformation
                                              (:parameterInformation
                                               (:labelOffsetSupport t)
                                               :activeParameterSupport t))
                        :references
                        (:dynamicRegistration :json-false)
                        :definition
                        (:dynamicRegistration :json-false :linkSupport t)
                        :declaration
                        (:dynamicRegistration :json-false :linkSupport t)
                        :implementation
                        (:dynamicRegistration :json-false :linkSupport t)
                        :typeDefinition
                        (:dynamicRegistration :json-false :linkSupport t)
                        :documentSymbol
                        (:dynamicRegistration :json-false :hierarchicalDocumentSymbolSupport t :symbolKind
                                              (:valueSet
                                               [1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26]))
                        :documentHighlight
                        (:dynamicRegistration :json-false)
                        :codeAction
                        (:dynamicRegistration :json-false :codeActionLiteralSupport
                                              (:codeActionKind
                                               (:valueSet
                                                ["quickfix" "refactor" "refactor.extract" "refactor.inline" "refactor.rewrite" "source" "source.organizeImports"]))
                                              :isPreferredSupport t)
                        :formatting
                        (:dynamicRegistration :json-false)
                        :rangeFormatting
                        (:dynamicRegistration :json-false)
                        :rename
                        (:dynamicRegistration :json-false)
                        :inlayHint
                        (:dynamicRegistration :json-false)
                        :publishDiagnostics
                        (:relatedInformation :json-false :codeDescriptionSupport :json-false :tagSupport
                                             (:valueSet
                                              [1 2])))
                       :window
                       (:workDoneProgress t)
                       :general
                       (:positionEncodings
                        ["utf-32" "utf-8" "utf-16"])
                       :experimental #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
                                                   ()))
                      :workspaceFolders
                      [(:uri "file:///home/USER/projects/csharp/test" :name "~/projects/csharp/test/")]))
[stderr] [22:21:00.959 INF] [Initialization] initializing, csharp-ls version 0.15.0.0
[server-notification] Sun Dec 22 22:21:00 2024:
(:jsonrpc "2.0" :method "window/showMessage" :params
          (:type 3 :message "csharp-ls: initializing, version 0.15.0.0"))
[stderr] [22:21:00.981 INF] [Initialization] csharp-ls is released under MIT license and is not affiliated with Microsoft Corp.; see https://github.com/razzmatazz/csharp-language-server
[stderr] [22:21:00.988 INF] [Initialization] MSBuildLocator: will register ".NET Core SDK", Version=8.0.404 as default instance
[server-notification] Sun Dec 22 22:21:00 2024:
(:jsonrpc "2.0" :method "window/showMessage" :params
          (:type 3 :message "csharp-ls: csharp-ls is released under MIT license and is not affiliated with Microsoft Corp.; see https://github.com/razzmatazz/csharp-language-server"))
[server-reply] (id:1) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 1 :result
          (:capabilities
           (:textDocumentSync
            (:openClose t :change 2 :save
                        (:includeText t))
            :completionProvider
            (:triggerCharacters
             ["." "'"])
            :hoverProvider t :signatureHelpProvider
            (:triggerCharacters
             ["(" "," "<" "{" "["])
            :definitionProvider t :typeDefinitionProvider t :implementationProvider t :referencesProvider t :documentHighlightProvider t :documentSymbolProvider t :codeActionProvider
            (:resolveProvider t)
            :codeLensProvider
            (:resolveProvider t)
            :workspaceSymbolProvider t :documentFormattingProvider t :documentRangeFormattingProvider t :documentOnTypeFormattingProvider
            (:firstTriggerCharacter ";" :moreTriggerCharacter
                                    ["}" ")"])
            :renameProvider t :callHierarchyProvider t :semanticTokensProvider
            (:legend
             (:tokenTypes
              ["class" "comment" "property" "enumMember" "enum" "event" "method" "variable" "interface" "keyword" "namespace" "number" "operator" "parameter" "struct" "regex" "string" "typeParameter"]
              :tokenModifiers
              ["static"])
             :range t :full t)
            :typeHierarchyProvider t :inlayHintProvider
            (:resolveProvider :json-false)
            :diagnosticProvider
            (:documentSelector
             [(:scheme "file" :pattern "**/*.cs")]
             :interFileDependencies :json-false :workspaceDiagnostics :json-false))
           :serverInfo
           (:name "csharp-ls" :version "0.15.0.0")))
[client-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "initialized" :params #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
                                                            ()))
[client-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "textDocument/didOpen" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs" :version 0 :languageId "csharp" :text "// See https://aka.ms/new-console-template for more information\nConsole.WriteLine(\"Hello, World!\");\n\n")))
[client-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "workspace/didChangeConfiguration" :params
          (:settings #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
                                   ())))
[client-request] (id:2) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 2 :method "textDocument/inlayHint" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :range
           (:start
            (:line 0 :character 0)
            :end
            (:line 3 :character 0))))
[stderr] [22:21:01.057 VRB] [Initialization] handleInitialized: "initialized" notification received from client
[server-request] (id:2) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 2 :method "client/registerCapability" :params
          (:registrations
           [(:id "116f9fd6-96b0-4a93-889d-681dafd5707c" :method "workspace/didChangeWatchedFiles" :registerOptions
                 (:watchers
                  [(:globPattern "**/*.{cs,csproj,sln}" :kind 7)]))]))
[client-reply] (id:2) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 2 :result nil)
[server-request] (id:3) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 3 :method "workspace/configuration" :params
          (:items
           [(:section "csharp")]))
[client-reply] (id:3) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 3 :result
          [nil])
[stderr] [22:21:01.096 VRB] [Initialization] handleInitialized: OK
[server-reply] (id:2) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 2 :result nil)
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "window/showMessage" :params
          (:type 3 :message "csharp-ls: attempting to find and load solution based on root path (\"/home/USER/projects/csharp/test\").."))
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "window/showMessage" :params
          (:type 3 :message "csharp-ls: 0 solution(s) found: []"))
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "window/showMessage" :params
          (:type 3 :message "csharp-ls: no or multiple .sln files found on /home/USER/projects/csharp/test"))
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "window/showMessage" :params
          (:type 3 :message "csharp-ls: looking for .csproj/fsproj files on /home/USER/projects/csharp/test.."))
[server-request] (id:4) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 4 :method "window/workDoneProgress/create" :params
          (:token "2e7352dc-44a1-4dff-81ba-5dfd8c446eba"))
[client-reply] (id:4) Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :id 4 :result nil)
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "$/progress" :params
          (:token "2e7352dc-44a1-4dff-81ba-5dfd8c446eba" :value
                  (:kind "begin" :title "Loading 1 project(s)..." :cancellable :json-false :message "0/1" :percentage 0)))
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "window/showMessage" :params
          (:type 3 :message "csharp-ls: loading project \"/home/USER/projects/csharp/test/test.csproj\".."))
[stderr] [22:21:01.723 ERR] [serverEventLoop] could not OpenProjectAsync('/home/USER/projects/csharp/test/test.csproj'): {exception}
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "$/progress" :params
          (:token "2e7352dc-44a1-4dff-81ba-5dfd8c446eba" :value
                  (:kind "report" :cancellable :json-false :message "test.csproj 1/1" :percentage 100)))
[server-notification] Sun Dec 22 22:21:01 2024:
(:jsonrpc "2.0" :method "$/progress" :params
          (:token "2e7352dc-44a1-4dff-81ba-5dfd8c446eba" :value
                  (:kind "end" :message "OK, 1 project file(s) loaded")))
[client-request] (id:3) Sun Dec 22 22:21:04 2024:
(:jsonrpc "2.0" :id 3 :method "textDocument/signatureHelp" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 0)))
[client-request] (id:4) Sun Dec 22 22:21:04 2024:
(:jsonrpc "2.0" :id 4 :method "textDocument/hover" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 0)))
[client-request] (id:5) Sun Dec 22 22:21:04 2024:
(:jsonrpc "2.0" :id 5 :method "textDocument/documentHighlight" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 0)))
[server-reply] (id:3) Sun Dec 22 22:21:04 2024:
(:jsonrpc "2.0" :id 3 :result nil)
[server-reply] (id:4) Sun Dec 22 22:21:04 2024:
(:jsonrpc "2.0" :id 4 :result nil)
[server-reply] (id:5) Sun Dec 22 22:21:04 2024:
(:jsonrpc "2.0" :id 5 :result nil)
[internal] (id:6) Sun Dec 22 22:21:12 2024:
(:deferring :textDocument/signatureHelp :id 6 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 1)))
[internal] (id:7) Sun Dec 22 22:21:12 2024:
(:deferring :textDocument/hover :id 7 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 1)))
[internal] (id:8) Sun Dec 22 22:21:12 2024:
(:deferring :textDocument/documentHighlight :id 8 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 1)))
[client-notification] Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :method "textDocument/didChange" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs" :version 1)
           :contentChanges
           [(:range
             (:start
              (:line 2 :character 0)
              :end
              (:line 2 :character 0))
             :rangeLength 0 :text "C")]))
[internal] Sun Dec 22 22:21:12 2024:
(:maybe-run-deferred
 (8 7 6))
[client-request] (id:8) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 8 :method "textDocument/documentHighlight" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 1)))
[client-request] (id:7) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 7 :method "textDocument/hover" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 1)))
[client-request] (id:6) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 6 :method "textDocument/signatureHelp" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 1)))
[client-request] (id:9) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 9 :method "textDocument/inlayHint" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :range
           (:start
            (:line 2 :character 0)
            :end
            (:line 3 :character 0))))
[server-reply] (id:8) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 8 :result nil)
[server-reply] (id:6) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 6 :result nil)
[server-reply] (id:7) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 7 :result nil)
[server-reply] (id:9) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 9 :result nil)
[internal] (id:10) Sun Dec 22 22:21:12 2024:
(:deferring :textDocument/signatureHelp :id 10 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 2)))
[internal] (id:11) Sun Dec 22 22:21:12 2024:
(:deferring :textDocument/hover :id 11 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 2)))
[internal] (id:12) Sun Dec 22 22:21:12 2024:
(:deferring :textDocument/documentHighlight :id 12 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 2)))
[client-notification] Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :method "textDocument/didChange" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs" :version 2)
           :contentChanges
           [(:range
             (:start
              (:line 2 :character 1)
              :end
              (:line 2 :character 1))
             :rangeLength 0 :text "o")]))
[internal] Sun Dec 22 22:21:12 2024:
(:maybe-run-deferred
 (12 11 10))
[client-request] (id:12) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 12 :method "textDocument/documentHighlight" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 2)))
[client-request] (id:11) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 11 :method "textDocument/hover" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 2)))
[client-request] (id:10) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 10 :method "textDocument/signatureHelp" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 2)))
[client-request] (id:13) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 13 :method "textDocument/inlayHint" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :range
           (:start
            (:line 2 :character 0)
            :end
            (:line 3 :character 0))))
[server-reply] (id:12) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 12 :result nil)
[server-reply] (id:10) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 10 :result nil)
[server-reply] (id:11) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 11 :result nil)
[server-reply] (id:13) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 13 :result nil)
[client-notification] Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :method "textDocument/didChange" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs" :version 3)
           :contentChanges
           [(:range
             (:start
              (:line 2 :character 2)
              :end
              (:line 2 :character 2))
             :rangeLength 0 :text "n")]))
[client-request] (id:14) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 14 :method "textDocument/completion" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 3)
           :context
           (:triggerKind 1)))
[server-reply] (id:14) Sun Dec 22 22:21:12 2024:
(:jsonrpc "2.0" :id 14 :result nil)
[client-request] (id:15) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 15 :method "textDocument/signatureHelp" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 3)))
[client-request] (id:16) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 16 :method "textDocument/hover" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 3)))
[client-request] (id:17) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 17 :method "textDocument/documentHighlight" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 3)))
[client-request] (id:18) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 18 :method "textDocument/inlayHint" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :range
           (:start
            (:line 2 :character 0)
            :end
            (:line 3 :character 0))))
[server-reply] (id:15) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 15 :result nil)
[server-reply] (id:16) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 16 :result nil)
[server-reply] (id:17) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 17 :result nil)
[server-reply] (id:18) Sun Dec 22 22:21:13 2024:
(:jsonrpc "2.0" :id 18 :result nil)
[client-notification] Sun Dec 22 22:21:15 2024:
(:jsonrpc "2.0" :method "textDocument/didChange" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs" :version 4)
           :contentChanges
           [(:range
             (:start
              (:line 2 :character 3)
              :end
              (:line 2 :character 3))
             :rangeLength 0 :text "s")]))
[client-request] (id:19) Sun Dec 22 22:21:15 2024:
(:jsonrpc "2.0" :id 19 :method "textDocument/completion" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 4)
           :context
           (:triggerKind 1)))
[server-reply] (id:19) Sun Dec 22 22:21:15 2024:
(:jsonrpc "2.0" :id 19 :result nil)
[client-notification] Sun Dec 22 22:21:15 2024:
(:jsonrpc "2.0" :method "textDocument/didChange" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs" :version 5)
           :contentChanges
           [(:range
             (:start
              (:line 2 :character 4)
              :end
              (:line 2 :character 4))
             :rangeLength 0 :text "o")]))
[client-request] (id:20) Sun Dec 22 22:21:15 2024:
(:jsonrpc "2.0" :id 20 :method "textDocument/completion" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 5)
           :context
           (:triggerKind 1)))
[server-reply] (id:20) Sun Dec 22 22:21:15 2024:
(:jsonrpc "2.0" :id 20 :result nil)
[client-request] (id:21) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 21 :method "textDocument/signatureHelp" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 5)))
[client-request] (id:22) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 22 :method "textDocument/hover" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 5)))
[client-request] (id:23) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 23 :method "textDocument/documentHighlight" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 5)))
[client-request] (id:24) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 24 :method "textDocument/inlayHint" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :range
           (:start
            (:line 2 :character 0)
            :end
            (:line 3 :character 0))))
[server-reply] (id:22) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 22 :result nil)
[server-reply] (id:21) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 21 :result nil)
[server-reply] (id:23) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 23 :result nil)
[server-reply] (id:24) Sun Dec 22 22:21:16 2024:
(:jsonrpc "2.0" :id 24 :result nil)
[client-notification] Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :method "textDocument/didChange" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs" :version 7)
           :contentChanges
           [(:range
             (:start
              (:line 2 :character 5)
              :end
              (:line 2 :character 5))
             :rangeLength 0 :text "l")
            (:range
             (:start
              (:line 2 :character 6)
              :end
              (:line 2 :character 6))
             :rangeLength 0 :text "e")]))
[client-request] (id:25) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 25 :method "textDocument/completion" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 7)
           :context
           (:triggerKind 1)))
[server-reply] (id:25) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 25 :result nil)
[client-request] (id:26) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 26 :method "textDocument/signatureHelp" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 7)))
[client-request] (id:27) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 27 :method "textDocument/hover" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 7)))
[client-request] (id:28) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 28 :method "textDocument/documentHighlight" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :position
           (:line 2 :character 7)))
[client-request] (id:29) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 29 :method "textDocument/inlayHint" :params
          (:textDocument
           (:uri "file:///home/USER/projects/csharp/test/Program.cs")
           :range
           (:start
            (:line 2 :character 0)
            :end
            (:line 3 :character 0))))
[server-reply] (id:28) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 28 :result nil)
[server-reply] (id:27) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 27 :result nil)
[server-reply] (id:26) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 26 :result nil)
[server-reply] (id:29) Sun Dec 22 22:21:17 2024:
(:jsonrpc "2.0" :id 29 :result nil)
[internal] (id:6) Sun Dec 22 22:21:22 2024:
(:timed-out :textDocument/signatureHelp :id 6 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 1)))
[internal] (id:7) Sun Dec 22 22:21:22 2024:
(:timed-out :textDocument/hover :id 7 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 1)))
[internal] (id:8) Sun Dec 22 22:21:22 2024:
(:timed-out :textDocument/documentHighlight :id 8 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 1)))
[internal] (id:10) Sun Dec 22 22:21:22 2024:
(:timed-out :textDocument/signatureHelp :id 10 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 2)))
[internal] (id:11) Sun Dec 22 22:21:22 2024:
(:timed-out :textDocument/hover :id 11 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 2)))
[internal] (id:12) Sun Dec 22 22:21:22 2024:
(:timed-out :textDocument/documentHighlight :id 12 :params
            (:textDocument
             (:uri "file:///home/USER/projects/csharp/test/Program.cs")
             :position
             (:line 2 :character 2)))

```

**Comment by razzmatazz** [comment]
Works for me on emacs-30 + doom emacs + csharp-ls 0.16.0 on macOS.

> <img width="1602" alt="Screenshot 2024-12-26 at 14 55 23" src="https://github.com/user-attachments/assets/7835cfe8-cbe7-4869-b3fa-3ad643dd9e4e" />

Let me try on bare emacs-29 + linux..

**Comment by bradprob** [comment]
@razzmatazz Thanks for your reply!

It seemed that a solution file was needed. The following steps solved my problem.

1. Create a console project with: `dotnet new console -n NameProject`
2. Go to the project directory and create a solution file: `cd NameProject && dotnet new sln -n NameProject`
3. Add the project to the solution file: `dotnet sln NameProject.sln add NameProject.csproj`

Eglot should work now with csharp-ls.

I close the ticket as my issue has been solved.

**Comment by RolandMarchand** [comment]
Personally, I had just installed `csharp-ls` to `~/.dotnet/tools` and had created the directory and added it to the PATH, but Emacs wasn't aware of the new PATH directory. I tried restarting Emacs, but same thing. I logged out and logged back in, and now the new PATH was available to my whole system, and Emacs was able to find `csharp-ls`. So if this LSP is the first .NET program you install, try to log out and log back in!
